third-party/cmake/DownloadAndExtractFile.cmake

```diff
@@ -22,7 +22,7 @@ set(SRC_DIR ${PREFIX}/src/${TARGET})
 
 # Check whether the source has been downloaded. If true, skip it.
 # Useful for external downloads like homebrew.
-if(EXISTS "${SRC_DIR}" AND IS_DIRECTORY "${SRC_DIR}")
+if(ALLOW_EXISTING_SRC_DIR AND EXISTS "${SRC_DIR}" AND IS_DIRECTORY "${SRC_DIR}")
```

Comment:
Kind of. This prevents that running `make deps` automatically builds updated dependencies. You can test it locally by changing some dependency version. CMake detects that CMakeLists has been changed, but it won't download the new version as the old source directory is still there.

Another problem with the cache is that it only stored the compiled output. As we don't know when a dependency has been updated and we have to re-run CMake, now I just cache the whole build directory and always run CMake. This makes the amount of cached data larger, though.

We could store only the compiled output and do something with timestamps, maybe compare the cache marker timestamp with the one from third-party/CMakeLists.txt and just rebuild everything if that has changed.. but I'm not sure if that would be such a good idea.